gcc/rust/checks/errors/rust-unsafe-checker.cc

```diff
@@ -60,7 +104,18 @@ UnsafeChecker::is_unsafe_context ()
 
 void
 UnsafeChecker::visit (IdentifierExpr &ident_expr)
-{}
+{
+  NodeId ast_node_id = ident_expr.get_mappings ().get_nodeid ();
+  NodeId ref_node_id;
+  HirId definition_id;
+
+  if (!resolver.lookup_resolved_name (ast_node_id, &ref_node_id))
```

Comment:
This is good, ideally this pass should be permissive of saw_errors eventually but we need to have the safe interface for the type context to return error nodes instead of nullptr to do that properly.